### PR TITLE
Support conditional execution of organized interactors

### LINKED ISSUE #128: Allow organized interactors to be skipped based on :if and :unless options

### PR DESCRIPTION
**Pull request by hedgesky**
Fixes #128.
I've decided to create `OrganizedInteractor` class to hold organized interactors along with passed options. This class is also responsible to run interactors with regard of passed options. `:if` and `:unless` are currently supported; you could pass either a symbol or a proc.

### CODE REVIEW

File: lib/interactor/organized_interactor.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  # options, such as :if, :unless, :before, :after
+  class OrganizedInteractor
+    attr_reader :interactor, :options
+
+    def initialize(interactor, options = {})
+      @interactor = interactor
+      @options = options
+    end
+
+    def call!(context, within_organizer)
```
Comment (hedgesky) [inline lib/interactor/organized_interactor.rb]: I'd rather use keyword argument for `within_organizer` here, but currently gemspec is targeted to Ruby 2.0.

**Comment by temkin** [comment]
This is a very good feature! Why not merged yet?

**Comment by gustavomazzoni** [comment]
+1

**Comment by hedgesky** [comment]
Hi @laserlemon, I know it's been a while since this conversation was started and probably now this gem isn't in your list of priorities. I'm not going to ask you to invest a lot of your time, but maybe you could merge the PR?
It will help people who want to use the feature. At the same time, leaving it undocumented makes it kind of experimental and leaves you with possibility to change it in future if you'd like to.

**Comment by irisbune** [comment]
+1 !

**Pull request closed without merging**